### PR TITLE
feat: remove the dependency on FusedMoE

### PR DESCRIPTION
**Pull request by zhyncs**
## Checklist

- [ ] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by zhyncs** [comment]
https://github.com/sgl-project/sglang/actions/runs/11995566939

**Comment by zhyncs** [comment]
Actually this is not a complete removal, just to avoid blocking @ispobock  from continuing development on DeepSeek V2 EP.

File: python/sglang/srt/utils.py [hunk 1]
```diff
@@ -930,3 +931,44 @@ def get_nvgpu_memory_capacity():
...
+        return torch.xpu.get_device_name(device_id)
+
+    if hasattr(torch, "hpu") and torch.hpu.is_available():
+        return torch.hpu.get_device_name(device_id)
+
+
+sglang_lib = Library("sglang", "FRAGMENT")  # noqa
+
+
+def direct_register_custom_op(
```
Comment (merrymercy) [inline python/sglang/srt/utils.py:953]: copy the docstring and mention its source.

File: python/sglang/srt/layers/triton_fused_moe/layer.py [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+
+from sglang.srt.layers.custom_op_util import register_custom_op
+from sglang.srt.layers.quantization.base_config import (
+    QuantizationConfig,
+    QuantizeMethodBase,
+)
+from sglang.srt.utils import set_weight_attrs
+
+if torch.cuda.is_available() or torch.hip.is_available():
+    from .fused_moe import fused_experts
```
Comment (merrymercy) [inline python/sglang/srt/layers/triton_fused_moe/layer.py:21]: use absolute path

File: python/sglang/srt/layers/triton_fused_moe/layer.py [hunk 3]
```diff
@@ -0,0 +1,631 @@
+from abc import abstractmethod
```
Comment (merrymercy) [inline python/sglang/srt/layers/triton_fused_moe/layer.py:1]: mention vllm source

File: python/sglang/srt/layers/triton_fused_moe/fused_moe.py [hunk 4]
```diff
@@ -0,0 +1,858 @@
+"""Fused MoE kernel."""
```
Comment (merrymercy) [inline python/sglang/srt/layers/triton_fused_moe/fused_moe.py:1]: mention vllm source

File: python/sglang/srt/layers/triton_fused_moe/configs/README [hunk 5]
```diff
@@ -0,0 +1,10 @@
+This directory contains tuned configurations for different settings of the fused_moe kernel.
```
Comment (merrymercy) [inline python/sglang/srt/layers/triton_fused_moe/configs/README:1]: 1. also copy the auto-tuned configs.
2. @ispobock you can probably auto-tune for some deepspeek models

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: python/sglang/srt/layers/triton_fused_moe/configs/README (hunk 5 above)
Comment (zhyncs) [reply python/sglang/srt/layers/triton_fused_moe/configs/README:1]: ref https://github.com/sgl-project/sglang/pull/2155

**Pull request merged**